Packet++/header/StpLayer.h

```diff
@@ -447,77 +600,151 @@ typedef stp_tcn_bpdu stp_header;
 		{
 		}
 
+		/**
+		 * Empty c'tor
+		 */
+		MultipleStpLayer();
+
 		/**
 		 * Get a pointer to Multiple STP header
 		 * @return A pointer to Multiple STP header
 		 */
 		mstp_conf_bpdu *getMstpHeader() const { return (mstp_conf_bpdu *)(m_Data); }
 
 		/**
-		 * @return Returns the length of version3 field.
+		 * Returns the length of version3 field
+		 * @return Length of version3 field
 		 */
 		uint16_t getVersion3Len() const;
 
+		/**
+		 * Sets the length of version3 field
+		 * @param[in] value Length of version3 field
+		 */
+		void setVersion3Len(uint16_t value);
+
 		/**
 		 * Returns the configuration ID format selector
 		 * @return Configuration ID of format selector
 		 */
 		uint8_t getMstConfigurationFormatSelector() const { return getMstpHeader()->mstConfigFormatSelector; }
 
 		/**
-		 * Returns the pointer to configuration name field.
+		 * Sets the configuration ID format selector
+		 * @param[in] value Configuration ID of format selector
+		 */
+		void setMstConfigurationFormatSelector(uint8_t value) { getMstpHeader()->mstConfigFormatSelector = value; }
+
+		/**
+		 * Returns the pointer to configuration name field
 		 * @return Configuration name
 		 */
 		std::string getMstConfigurationName() const;
 
+		/**
+		 * Sets the configuration name field
+		 * @param[in] value Configuration name. Length should be less than 32, if longer value provided first 32
+		 * characters are used
+		 */
+		void setMstConfigurationName(const std::string &value);
+
 		/**
 		 * Returns the revision of configuration ID
 		 * @return Revision of configuration ID
 		 */
-		uint16_t getMstConfigRevision() const { return getMstpHeader()->mstConfigRevision; }
+		uint16_t getMstConfigRevision() const;
```

Comment:
For information; noticed a bug here. It should be in big endian so moved to cpp.